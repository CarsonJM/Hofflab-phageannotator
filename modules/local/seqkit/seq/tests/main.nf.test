nextflow_process {

    name "Test Process: SEQKIT_SEQ"
    script "../main.nf"
    process "SEQKIT_SEQ"


    test("fasta.gz") {

        when {
            params {
                assembly_min_length = 1000
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            params {
                assembly_min_length = 1000
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
