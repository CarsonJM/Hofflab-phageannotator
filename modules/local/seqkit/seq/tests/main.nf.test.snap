{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ac420fb44c91527db6f8db702ac321de"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ac420fb44c91527db6f8db702ac321de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T08:08:57.910688485"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,483f4a5dfe60171c86ee9b7e6dff908b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ac420fb44c91527db6f8db702ac321de"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,483f4a5dfe60171c86ee9b7e6dff908b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ac420fb44c91527db6f8db702ac321de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T08:08:40.074388049"
    }
}